src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -994,12 +1051,18 @@ public bool StatusCanHaveBody(int statusCode)
                    statusCode != 304;
         }
 
-        public void ReportCorruptedHttpRequest(BadHttpRequestException ex)
+        public void ReportBadHttpRequest(BadHttpRequestException ex)
```

Comment:
Can you simplify the calling pattern by passing a string and constructing the exception internally?